tests/ignite/metrics/test_psnr.py

```diff
@@ -79,144 +79,344 @@ def test_psnr():
     _test_psnr(y_pred, y, data_range, device)
 
 
-def _test(
-    y_pred,
-    y,
-    data_range,
-    metric_device,
-    n_iters,
-    s,
-    offset,
-    rank,
-    atol,
-    output_transform=lambda x: x,
-    compute_y_channel=False,
-):
-    from ignite.engine import Engine
-
-    def update(engine, i):
-        return (
-            y_pred[i * s + offset * rank : (i + 1) * s + offset * rank],
-            y[i * s + offset * rank : (i + 1) * s + offset * rank],
-        )
+def _test_distrib_input_float(device, atol=1e-8):
+    def _test(
+        y_pred,
+        y,
+        data_range,
+        metric_device,
+        n_iters,
+        batch_size,
+        atol,
+        output_transform=lambda x: x,
+        compute_y_channel=False,
+    ):
```

Comment:
Let's try to eliminate duplicated code

Reply:
Do you mean to extract the `def _test` outside the test code and be used commonly from other test methods?

Reply:
Any solution work, just do not add duplicated code.